Support for agents authenticated to GIT using non-SSH in release changelog functionality

aly76 opened the pull request:
#514 - Add support for agents in the `orchestrator:release` command's `--generatechangelog` functionality using any other authentication mechanism other than SSH

**breaking** `repoUrl` no longer accepted in the release definition yaml 
**breaking** `--repoUrl` no longer a flag in `changelog:generate`. The command requires the cwd to be the source directory.

aly76 committed:
Fix release changelog for hosted agents

aly76 committed:
Fix ReleaseDefinition tests

azlam-abdulsalam committed:
Merge branch 'develop' into fix-changelog-auth

azlam-abdulsalam commented:
changelog:generate should provide an option of no push, so the user can push it to any repo

aly76 committed:
Remove repoUrl flag from changelog:generate

Align change:generate with orchestrator:release, so that it requires the CWD to
be the source directory

azlam-abdulsalam committed:
Merge branch 'develop' into fix-changelog-auth
